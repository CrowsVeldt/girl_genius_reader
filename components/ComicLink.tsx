import { useContext } from "react";
import { Text, TouchableOpacity } from "react-native";
import { StyleSheet } from "react-native";
import { DateContext } from "../context/DateContext";
import { formatDate } from "../utils/utilFunctions";

export default function ComicLink({ date, nav }: { date: string; nav: any }) {
  const { changeCurrentDate } = useContext(DateContext);
  return (
    <TouchableOpacity
      style={styles.linkButton}
      onPress={() => {
        changeCurrentDate(date);
        nav.navigate("Home");
      }}
    >
      <Text style={styles.linkText}>{formatDate(date)}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  linkButton: {
    borderColor: "black",
    borderWidth: 1,
  },
  linkText: {
    textAlign: "center",
  },
});
