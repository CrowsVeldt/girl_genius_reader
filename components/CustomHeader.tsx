import {
  View,
  Text,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
} from "react-native";
import StarButton from "./StarButton";
import { useContext } from "react";
import { ComicContext } from "../context/ComicContext";

export default function CustomHeader({
  navigation,
  route,
  options,
  layout,
}: {
  navigation: any;
  route: any;
  options: any;
  layout: any;
}) {
  const { getCurrentPage } = useContext(ComicContext);

  const page = getCurrentPage();
  const routeName = route.name;

  return (
    <View style={[options.headerStyle, styles.header]}>
      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.openDrawer()}
      >
        <Text style={styles.buttonText}>â‰¡</Text>
      </TouchableOpacity>
      <Text style={styles.headerTitle}>
        {routeName === "Home"
          ? `Volume ${page.volume}, Page ${page.pageNumber}`
          : routeName}
      </Text>
      {routeName === "Home" && <StarButton page={page} />}
      {routeName !== "Home" && <View style={styles.placeholderBox}></View>}
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: StatusBar.currentHeight,
  },
  headerTitle: {
    textAlignVertical: "center",
    textAlign: "center",
    fontSize: 18,
    flexGrow: 1,
  },
  button: {
    backgroundColor: process.env.EXPO_PUBLIC_DARK_BG_COLOR,
    width: 50,
    height: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonText: {
    fontSize: 30,
  },
  placeholderBox: {
    height: 40,
    width: 40,
    marginRight: 10,
  },
});
