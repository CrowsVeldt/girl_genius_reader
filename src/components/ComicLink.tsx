import { ContextType, useContext } from "react";
import { StyleSheet, Text, TouchableOpacity } from "react-native";
import { formatDate } from "../utils/utilFunctions";
import { PageType } from "../utils/types";
import { ComicContext } from "../context/ComicContext";

export default function ComicLink({ page, nav }: { page: PageType; nav: any }) {
  const { changeCurrentPage }: ContextType<typeof ComicContext> =
    useContext(ComicContext);
  return (
    <TouchableOpacity
      style={styles.linkButton}
      onPress={() => {
        changeCurrentPage(page);
        nav.navigate("ComicPage");
      }}
    >
      <Text style={styles.linkNumber}>{page.pageNumber}</Text>
      <Text style={styles.linkText}>{formatDate(page.date)}</Text>
      <Text style={styles.title}>{page.title}</Text>
    </TouchableOpacity>
  );
}

const styles = StyleSheet.create({
  linkButton: {
    width: "100%",
    height: 40,
    flexDirection: "row",
    justifyContent: "space-between",
    borderBottomColor: "black",
    borderBottomWidth: 1,
  },
  linkNumber: {
    marginRight: 6,
    flex: 1,
  },
  linkText: {
    flex: 2,
  },
  title: {
    flex: 3,
    overflow: "visible",
  },
});
