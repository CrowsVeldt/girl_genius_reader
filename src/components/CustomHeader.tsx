import { ContextType, useContext } from "react";
import {
  View,
  Text,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
} from "react-native";
import { PageType } from "../utils/types";
import { ComicContext } from "../context/ComicContext";
import PullToRefresh from "./PullToRefresh";
import StarButton from "./StarButton";

export default function CustomHeader({
  navigation,
  route,
  options,
  layout,
}: {
  navigation: any;
  route: any;
  options: any;
  layout: any;
}) {
  const { getCurrentPage }: ContextType<typeof ComicContext> =
    useContext(ComicContext);

  const page: PageType = getCurrentPage();
  const routeName: string = route.name;

  return (
    <View style={[options.headerStyle, styles.header]}>
      <PullToRefresh />
      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.openDrawer()}
      >
        <Text style={styles.buttonText}>â‰¡</Text>
      </TouchableOpacity>
      <Text style={styles.headerTitle}>
        {routeName === "Home"
          ? `Volume ${page.volumeNumber} - Page ${page.pageNumber}`
          : routeName}
      </Text>
      {routeName === "Home" && <StarButton page={page} />}
      {routeName !== "Home" && <View style={styles.placeholderBox}></View>}
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: StatusBar.currentHeight,
    width: "100%",
  },
  headerTitle: {
    textAlignVertical: "center",
    textAlign: "center",
    fontSize: 18,
    flexGrow: 1,
  },
  button: {
    backgroundColor: process.env.EXPO_PUBLIC_DARK_BG_COLOR,
    width: 50,
    height: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonText: {
    fontSize: 30,
  },
  placeholderBox: {
    height: 40,
    width: 40,
    marginRight: 10,
  },
});
