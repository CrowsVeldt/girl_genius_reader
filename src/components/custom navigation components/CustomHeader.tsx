import { ContextType, useContext } from "react";
import {
  View,
  Text,
  StyleSheet,
  StatusBar,
  TouchableOpacity,
} from "react-native";
import { PageType } from "../../utils/types";
import { ComicContext } from "../../context/ComicContext";
import PullToRefresh from "../gesture components/PullToRefresh";
import BookmarkButton from "../control components/BookmarkButton";
import { router } from "expo-router";

const names: any = {
  index: "Home",
  bookmarks: "Bookmarks",
  privacy: "Privacy",
  acknowledgments: "Acknowledgments",
  options: "Options",
  message: "Message",
  changelog: "Changelog",
};

export default function CustomHeader({
  navigation,
  route,
  options,
  layout,
}: {
  navigation: any;
  route: any;
  options: any;
  layout: any;
}) {
  const {
    getCurrentPage,
    getCurrentVolume,
    changeCurrentVolume,
  }: ContextType<typeof ComicContext> = useContext(ComicContext);

  const page: PageType = getCurrentPage();
  const volumeNumber: number = getCurrentVolume();
  const routeName: string = route.name;

  return (
    <View style={[options.headerStyle, styles.header]}>
      <PullToRefresh />
      <TouchableOpacity
        style={styles.button}
        onPress={() => navigation.openDrawer()}
      >
        <Text style={styles.buttonText}>â‰¡</Text>
      </TouchableOpacity>
      {routeName === "comicpage" && (
        <View style={styles.volumeLink}>
          <TouchableOpacity
            onPress={() => {
              changeCurrentVolume(page.volumeNumber);
              router.push("volume");
            }}
          >
            <Text style={styles.headerTitle}>
              {`Volume ${page.volumeNumber}`}
            </Text>
          </TouchableOpacity>
          <Text style={styles.headerTitle}>{` - Page ${page.pageNumber}`}</Text>
        </View>
      )}
      {routeName === "volume" && (
        <Text style={styles.headerTitle}>{`Volume ${volumeNumber}`}</Text>
      )}
      {routeName !== "comicpage" && routeName !== "volume" && (
        <Text style={styles.headerTitle}>{names[routeName]}</Text>
      )}
      {routeName === "comicpage" && <BookmarkButton page={page} />}
      {routeName !== "comicpage" && <View style={styles.placeholderBox}></View>}
    </View>
  );
}

const styles = StyleSheet.create({
  header: {
    flexDirection: "row",
    alignItems: "center",
    justifyContent: "space-between",
    marginTop: StatusBar.currentHeight,
    width: "100%",
    height: 60,
    borderBottomColor: "black",
    borderBottomWidth: 1
  },
  headerTitle: {
    textAlignVertical: "center",
    textAlign: "center",
    fontSize: 18,
  },
  volumeLink: {
    flexDirection: "row",
  },
  button: {
    backgroundColor: process.env.EXPO_PUBLIC_DARK_BG_COLOR,
    width: 50,
    height: "100%",
    alignItems: "center",
    justifyContent: "center",
  },
  buttonText: {
    fontSize: 30,
  },
  placeholderBox: {
    height: 40,
    width: 40,
    marginRight: 10,
  },
});
