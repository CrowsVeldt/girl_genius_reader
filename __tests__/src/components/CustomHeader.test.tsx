import renderer from "react-test-renderer";
import ComicLink from "../../../src/components/CustomHeader";
import ComicProvider from "../../../src/context/ComicContext";
import CustomHeader from "../../../src/components/CustomHeader";

describe("<ComicLink />", () => {
  const testPage = {
    date: "20021104",
    title: "",
    pageNumber: 1,
    volume: 1,
  };

  it("renders the same as last time", () => {
    const tree = renderer.create(
      <ComicProvider>
        <CustomHeader
          navigation={jest.mock}
          route={jest.mock}
          options={jest.mock}
          layout={jest.mock}
        />
      </ComicProvider>
    );
    expect(tree).toMatchSnapshot();
  });
});
