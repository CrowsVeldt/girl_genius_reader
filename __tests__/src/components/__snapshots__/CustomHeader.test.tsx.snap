// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ComicLink /> renders the same as last time 1`] = `
<View
  style={
    [
      undefined,
      {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginTop": null,
        "width": "100%",
      },
    ]
  }
>
  <View
    accessibilityState={
      {
        "busy": undefined,
        "checked": undefined,
        "disabled": undefined,
        "expanded": undefined,
        "selected": undefined,
      }
    }
    accessibilityValue={
      {
        "max": undefined,
        "min": undefined,
        "now": undefined,
        "text": undefined,
      }
    }
    accessible={true}
    collapsable={false}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      {
        "alignItems": "center",
        "backgroundColor": undefined,
        "height": "100%",
        "justifyContent": "center",
        "opacity": 1,
        "width": 50,
      }
    }
  >
    <Text
      style={
        {
          "fontSize": 30,
        }
      }
    >
      â‰¡
    </Text>
  </View>
  <Text
    style={
      {
        "flexGrow": 1,
        "fontSize": 18,
        "textAlign": "center",
        "textAlignVertical": "center",
      }
    }
  >
    mock
  </Text>
  <View
    style={
      {
        "height": 40,
        "marginRight": 10,
        "width": 40,
      }
    }
  />
</View>
`;
